DQM/EcalMonitorTasks/interface/TimingTask.h

```diff
@@ -19,10 +19,18 @@ namespace ecaldqm {
     void runOnRecHits(EcalRecHitCollection const&, Collections);
     void runOnUncalibRecHits(EcalUncalibratedRecHitCollection const&);
 
+    enum Constants {
+      nBXBins = 15
+    };
+
   private:
     void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+    void beginEvent(edm::Event const&, edm::EventSetup const&) override;
     void setParams(edm::ParameterSet const&) override;
 
+    int bxBinEdges_[nBXBins + 1];
```

Comment:
did this not work?

int* pBin(std::upper_bound(bxBinEdges_.begin(), bxBinEdges_.end(), _evt.bunchCrossing()));